### ISSUE TITLE
4 button label strings + notification email subject not properly escaped

### ISSUE THREAD

**Issue by FlyThePlanet** [ultimatemember/ultimatemember#453]
### Ultimate Member Version
Tell us what UM core version you use. 
- [x] 2.0.16-2.0.56 ( latest )

### Subject of the issue
The language strings of the following button labels are not properly escaped:
- Default registration form > Registration Primary Button
- Default registration form > Registration Secondary Button
- Default login form > Login Primary Button
- Default login form > Login Secondary Button

Same for the subject of the notification emails.

### Expected behavior
Labels and email subject that include simple quotes should be properly displayed

### Actual behavior
Labels and email subject that include simple quotes are not properly displayed
Example: "S'enregistrer" (in French) is displayed "S\\'enregistrer"
Example: "Compte en attente d'activation" (in French) is displayed "Compte en attente d\\'activation"

### Steps to reproduce the behavior
Set a string like "S'enregistrer" in Ultimate Member > Settings > Appearance > Registration form > "Login Primary Button Text" and check how it's displayed on the registration form.

Set a string like "Compte en attente d'activation" in Ultimate Member > Settings > Email > Edit > Subject and check how it's displayed when you receive the email.

## PHP version
- [x] 7.1 above
- [ ] 5.4
- [ ] 5.3 below

### Do you use UM extensions? 
- [ ] Private Content 
- [ ] Instagram
- [ ] User Tags 
- [ ] Social Activity
- [ ] WooCommerce
- [ ] Private Messages
- [ ] Followers
- [ ] Real-time Notifications
- [ ] Social Login
- [ ] bbPress
- [ ] MailChimp
- [ ] User Reviews
- [ ] Verified Users
- [ ] myCRED
- [ ] Notices
- [ ] Profile Completeness 
- [ ] Friends
- [ ] Terms & Conditions
- [x] Google reCAPTCHA 
- [ ] Online Users

**Comment by FlyThePlanet** [comment]
@champsupertramp, @nikitasinelnikov any chance this issue will be fixed anytime soon?
Should be quite easy to fix.
Thanks!

**Comment by nikitasinelnikov** [comment]
hi @FlyThePlanet 

Please upgrade Ultimate Member to the latest 2.0.50 version, which is available to download from "master" branch here or here https://wordpress.org/plugins/ultimate-member/

Let me know if this works after an upgrade,
Thanks!

**Comment by FlyThePlanet** [comment]
Hi @nikitasinelnikov 

Sorry for the late reply. I was overseas.

Ok, so with the last version (2.0.56)...
- it's fixed for the 4 buttons,
- but it's NOT fixed for the email subjects!

Just try: Ultimate Member > Settings > Email > Your account is pending review > Edit > Subject > enter the following subject "You're nearly done! Your account is now pending review" and check the email subjet displayed when you'll receive the email.

You should see "You\'re nearly done! Your account is now pending review" with an "\" before the apostrophe.

P.S. for UM users: the temporary workaround awaiting this issue is fixed is to enter the subject using `&#39;` instead of apostrophes.

**Comment by FlyThePlanet** [comment]
Hi @nikitasinelnikov and @champsupertramp 

Any news on this issue?
It should be easy to fix.
It's fixed for the button labels but not for the email subjects.

It's the last bug I have with UM :-p

Thanks!

**Comment by champsupertramp** [comment]
Hi @FlyThePlanet,

We are not able to reproduce the issue with the steps above. Here's a screenshot of the Email:

<img width="1171" alt="Screen Shot 2019-09-08 at 2 29 11 AM" src="https://user-images.githubusercontent.com/12200688/64478806-78bc5a00-d1e0-11e9-8b1d-968eff56db7c.png">

Are you using any Mail plugin?

Regards,

**Comment by champsupertramp** [comment]
Here's the Email template settings:
<img width="1254" alt="Screen Shot 2019-09-08 at 2 31 50 AM" src="https://user-images.githubusercontent.com/12200688/64478827-de104b00-d1e0-11e9-930c-9b4aa7bef2f1.png">


Regards,

**Comment by FlyThePlanet** [comment]
Hi @champsupertramp 

> Are you using any Mail plugin?

Nope. Just receive the email in Gmail.

Here is a new test on a new template... (but it's the same for all templates)
![image](https://user-images.githubusercontent.com/39710808/64485999-c14b3600-d227-11e9-968f-754e2e28a0da.png)

And the email received in Gmail...
![image](https://user-images.githubusercontent.com/39710808/64486162-d0cb7e80-d229-11e9-9e61-e0a8e44d5da9.png)

The apostrophe character is the ANSI 27/Unicode 0027.

**Comment by FlyThePlanet** [comment]
Question: where are those email subjet strings are saved in the DB?

**Comment by champsupertramp** [comment]
Hi @FlyThePlanet, Please try dumping the email subject with the code below. Please paste the output here.

```
$pending_email_sub = UM()->options()->get( 'pending_email_sub' );
$subject =  wp_unslash( $subject );
var_dump( $subject ); 
```

The above code snippet is how we populate the mail subject in:
https://github.com/ultimatemember/ultimatemember/blob/master/includes/core/class-mail.php#L437-L438

Regards,

**Comment by FlyThePlanet** [comment]
Hi @champsupertramp 

> Please try dumping the email subject with the code below.

What do you expect me to do?
Copy this code right after the line 438 and copy/paste the content of the variable that should be displayed after the submit of the registration form?

If yes, it doesn't work. The content of the variable is not displayed on the screen right after the submit.

**Comment by FlyThePlanet** [comment]
Ok I used file_put_contents().
Here is the content: "Compte en attente d'activation" without the "\"

**Comment by champsupertramp** [comment]
Hi @FlyThePlanet ,

Thanks for letting us know how you resolved this issue. We don't have any similar issue reported to us. We never got this issue reported to our support system in ultimatemember.com as well.

I am closing this issue  now.

Regards,

**Comment by FlyThePlanet** [comment]
> Thanks for letting us know how you resolved this issue.

@champsupertramp I did not resolve anything.
I said "YOU found the WAY to fix the issue".

Your code works for a variable.
You probably just need to apply it to the code building the email.

So far nothing is fixed.

> We don't have any similar issue reported to us. We never got this issue
> reported to our support system in ultimatemember.com as well.

But let's start by the beginning, when you enter a subject like "You're nearly done! Your account is now pending review", what do you have in the DB?
"You're nearly done! Your account is now pending review or "You\'re nearly done! Your account is now pending review"?

**Comment by FlyThePlanet** [comment]
Hi @champsupertramp 

Sorry, I was not at home yesterday.

> I have this temporary fixes that you can add in the theme's functions.php file

Thanks!
I confirm, it works :+1: 
At least there's a temp fix.

> i have the `You\'re nearly done! Your account is now pending review` in the database.

Ok. It's same as me then.
So when you set `You're nearly done! Your account is now pending review` as email subject in WP > Ultimate Member > Settings > Email > edit one email > Subject, it's saved `You\'re nearly done! Your account is now pending review` for everyone.

Now when you run the test you requested here...
https://github.com/ultimatemember/ultimatemember/issues/453#issuecomment-530345153
...what do you have? The subject with or without the anti-slash? I still have it.

**Comment by champsupertramp** [comment]
Hi @FlyThePlanet ,

The code snippet in https://github.com/ultimatemember/ultimatemember/issues/453#issuecomment-530345153 won't be affected by the temp fix. Please disregard it as i wanted to see the dumped data to investigate the issue.

Please see fixes added for RC: https://github.com/ultimatemember/ultimatemember/commit/2629eb4ace97da9fbc7e922f38d06f1f370c4f64


Regards,

**Comment by FlyThePlanet** [comment]
Hi @champsupertramp 

> The code snippet in #453 (comment) won't be affected by the temp fix. Please disregard it as i wanted to see the dumped data to investigate the issue.

I know. But my point was you said `We are not able to reproduce the issue` but you also observed that you indeed had `You\'re nearly done!` with the anti-slash in the DB. So it meant even for you, you had the anti-slash in DB and it was removed somewhere in the process of sending the email, while it was not for me. So I was also investigating, trying to see if for you at the step of the variable it was still there.

> Please see fixes added for RC: 2629eb4

Yes, it works! :+1: 
Thanks again @champsupertramp 

But have you found why/where the anti-slash (while it was in your DB) was removed w/o the fix for you?
